### PR TITLE
[plsql] Parse Exception when using '||' operator in where clause

### LINKED ISSUE #1507: [plsql] Parse Exception when using '||' operator in where clause
**Affects PMD Version:** 6.10.0-SNAPSHOT

**Description:**

**Code Sample demonstrating the issue:**

```
DECLARE
BEGIN

SELECT *
  INTO x
  FROM y
  WHERE a = 'a' || 'b';

END;
/
```

Exception:

```
net.sourceforge.pmd.lang.plsql.ast.ParseException: Encountered " "||" "|| "" at line 7, column 17.
Was expecting one of:
    ";" ...
    "FETCH" ...
    "GROUP" ...
    "OFFSET" ...
    "ORDER" ...
    
	at net.sourceforge.pmd.lang.plsql.ast.PLSQLParser.generateParseException(PLSQLParser.java:53972)
	at net.sourceforge.pmd.lang.plsql.ast.PLSQLParser.jj_consume_token(PLSQLParser.java:53823)
	at net.sourceforge.pmd.lang.plsql.ast.PLSQLParser.UnlabelledStatement(PLSQLParser.java:15265)
	at net.sourceforge.pmd.lang.plsql.ast.PLSQLParser.Statement(PLSQLParser.java:15980)
	at net.sourceforge.pmd.lang.plsql.ast.PLSQLParser.Block(PLSQLParser.java:848)
	at net.sourceforge.pmd.lang.plsql.ast.PLSQLParser.Global(PLSQLParser.java:497)
	at net.sourceforge.pmd.lang.plsql.ast.PLSQLParser.Input(PLSQLParser.java:194)
	at net.sourceforge.pmd.lang.plsql.PLSQLParser.parse(PLSQLParser.java:49)
	at net.sourceforge.pmd.util.fxdesigner.model.ASTManager.updateIfChanged(ASTManager.java:93)
...
```

**Running PMD through:** Designer

### PR DESCRIPTION
**Pull request by adangel**
Fixes #1507

### CODE REVIEW

**Comment by pmd-test** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="true">No java rules are changed!</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: pmd-plsql/etc/grammar/PldocAST.jjt [hunk 1]
```diff
@@ -1414,7 +1414,7 @@ ASTSqlExpression SqlExpression() :
  |
     LOOKAHEAD(2) <ROWNUM>
  |
-    LOOKAHEAD(2) Literal()
+    LOOKAHEAD(2) AdditiveExpression() // this can be a literal or a simple expression, but no conditional
```
Comment (oowekyala) [inline pmd-plsql/etc/grammar/PldocAST.jjt]: > this can be a literal or a simple expression, but no conditional

Maybe we should add a test asserting that using a conditional fails? I'm not sure whether we have negative parsing tests yet. Maybe it's worth it also to remove the duplication of testing logic and make it easier to add new tests. Or alternatively, use pmd/pmd-regression-tester to parse real-world PLSQL

Comment (adangel) [reply pmd-plsql/etc/grammar/PldocAST.jjt]: the comment is more a reminder for me...
I'm actually taking a short-cut here, reusing the plsql-expression in a sql-expression - the sql-expression is more restrictive, that's why we can't use "Expression()" here.

I don't think, we need an extra (negative) parsing test here. I might change this SqlExpression in the future.
Using pmd-regression-tester is definitively what we should use.
